cmd/erasure-server-pool-decom.go

```diff
@@ -842,6 +838,22 @@ func (z *erasureServerPools) decommissionPool(ctx context.Context, idx int, pool
 						continue
 					}
 					stopFn := globalDecommissionMetrics.log(decomMetricDecommissionObject, idx, bi.Name, version.Name, version.VersionID)
+					if version.IsRemote() {
+						gr.Close()
```

Comment:
why should there is a GetObject() call for a tiered object? while decommissioning? @poornas we don't have to do `gr.Close()` in such a scenario.

Reply:
[line](https://github.com/minio/minio/pull/16751/files#diff-1947f14a42e1b32588f855644d15ad4705512304009e7cc3e82411a30cd630a2L824) above already called GetObjectNInfo - the reader needs to be closed to avoid leak since code path does not go thru decommissionObject (which closes the reader)

Reply:
Yeah but why do you need to have the reader, for a tiered object we don't need to send a GetObject()